pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -756,80 +764,105 @@ def _handle_distribution_tree(declarative_content):
 
             elif isinstance(declarative_content.content, Modulemd):
                 for pkg in declarative_content.extra_data['package_relation']:
-                    try:
-                        with transaction.atomic():
-                            declarative_content.content.packages.add(pkg.content)
-                    except ValueError:
-                        pass
+                    if pkg.content.pk and declarative_content.content.pk:
```

Comment:
Checking `pk` doesn't work for models with UUIDs. You should check `_state` instead:

```
In [1]: repository = Repository()

In [2]: repository.pk
Out[2]: UUID('ea70e2e6-4427-4242-b60e-85f04b4cdc92')

In [3]: repository._state.adding
Out[3]: True
```